src/controllers/transactions.js

```diff
@@ -1,62 +1,119 @@
 const service = require('../services/transactions');
 
-// Aby odpowiedzi były spójne daje przykładowy res:
-
-//   res.json({
-//       status: 200,
-//       statusText: "OK",
-//       data: {
-//         ...
-//       },
-//     });
-
 const get = async (req, res, next) => {
   const { _id } = req.user;
   try {
     const transactions = await service.getTransactions({ _id });
     res.json({ status: 200, statusText: 'OK', data: transactions });
   } catch (err) {
-    res.status(500).json({status: 500, statusText: "Internal serwer error", error:err})
     next(err);
   }
 };
 
 const getById = async (req, res, next) => {
+  const { id } = req.params;
+  const userId = req.user._id;
   try {
+    const result = await service.getTransactionById(userId, id);
+    if (result) {
+      return res.json({
+        status: 200,
+        statusText: 'OK',
+        data: { result },
+      });
+    }
+    res.status(404).json({
+      status: 404,
+      statusText: 'Not Found',
+      data: `Not found transaction id: ${id}`,
+    });
   } catch (err) {
-    console.error(err.message);
     next(err);
   }
 };
 
 const getCategory = async (req, res, next) => {
+  const { id } = req.params;
+  const userId = req.user._id;
+  const transaction = await service.getTransactionById(userId, id);
+  if (!transaction) {
+    return res.status(404).json({
+      status: '404',
+      statusText: 'Not Found',
+      data: `Not found transaction id: ${id}`,
+    });
+  }
   try {
+    const result = await service.getTransactionCategory(userId, id);
+    res.json(result);
   } catch (err) {
     console.error(err.message);
     next(err);
   }
 };
```

Comment:
Tutaj getTransactionCategory jak nie znajdzie kategorii to w sumie znaczy że nie znalazł transakcji bo tak samo wyszukuje jak getTransactionById, dlatego tamto wyżej niepotrzebne i ten if sprawdzić wtedy dla result. Res mógłby być jak inne czyli status statusText data dla spójności i tu w catch log error nie usunąłeś jak wszędzie indziej